docs/architecture/adr-111-nil-mempool.md

```diff
@@ -0,0 +1,239 @@
+    what to do with the transaction list coming in `RequestPrepareProposal`,
+    whether it wants to maintain an app-side mempool (more on this below), and whether or not
+    to combine the transactions in the app-side mempool with those coming in `RequestPrepareProposal`
+  * all those combinations are hard to fully understand, as the semantics and guarantees are
+    often not clear
+  * when using exclusively an app-mempool (the approach taken in the Cosmos SDK `v0.47.x`)
+    for populating proposed blocks, with the aim of simplifying the app developers' life,
+    we still have a suboptimal model where we need to continue using CometBFT's mempool
+    in order to disseminate the transactions. So, we end up using twice as much memory,
+    as in-transit transactions need to be kept in both mempools.
+
+The approach presented in this ADR builds on the app-mempool design released in `v0.47.x`
+of the Cosmos SDK,
+and briefly discussed in the last bullet point above (see [SDK app-mempool][sdk-app-mempool] for further details of this model).
+
+In the app-mempool design in Cosmos SDK `v0.47.x`
+an unconfirmed transaction must be both in CometBFT's mempool for dissemination and
+in the app's mempool so the application can decide how to manage the mempool.
+The many advantages of this approach are beyond question. However it has some implications:
+
+* Having every transaction both in CometBFT and in the application is suboptimal in terms of memory.
+  Additonally, the app developer has to be careful
+  that the contents of both mempools does not diverge over time
+  (hence the crucial role `re-CheckTx` plays post-ABCI++).
+* The main reason for a transaction needing to be in CometBFT's mempool is
+  because the design in Cosmos SDK `v0.47.x` does not consider an application
+  that has its own means of disseminating transactions.
+  It reuses the peer to peer network underneath CometBFT reactors.
+* So, if an app has an ad-hoc design of how to disseminate transactions,
+  there is no point in having transactions in CometBFT's mempool.
+
+This proposal targets this kind of applications:
+those that have an ad-hoc (and likely more efficient) mechanism for transaction dissemination.
+We propose to introduce a `nil` mempool which will effectively act as a stubbed object
+internally:
+
+* it will reject any transaction being locally submitted or gossipped by a peer
+* when it is time to _reap_ (as it is currently called) the mempool the answer will always be empty
+* the application running on the proposer validator will add transactions it received
+  using the appchains's own mechanism via `PrepareProposal`.
+
+## Alternative Approaches
+
+These are the alternatives known to date:
+
+1. Keep the current model. Useful for basic apps, but clearly suboptimal for applications
+   with their own mechanism to disseminate transactions and particular performance requirements.
+2. Provide more efficient general-purpose mempool implementations.
```

Comment:
Should we mention the CAT mempool here ? It's a good example of a new design that has been some R&D, testing, validation to ensure a different approach is a better alternative